### PR TITLE
messaging: Add new modules for kafka-{acls,topics,configs}

### PR DESCRIPTION
**Pull request by gdelpierre**
##### SUMMARY
This module aims to allow a user to manage Kafka
  - ACLs
  - Topics
  - Configs  

Internally it relies on several kafka scripts.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
 - kafka_acls.py
 - kafka_configs.py
 - kafka_topics.py

##### ANSIBLE VERSION
```
ansible 2.3.1.0
  config file = /home/gdelpierre/devel/roles/ansible.cfg
  configured module search path = [u'./library/']
  python version = 2.7.13 (default, Jan 19 2017, 14:48:08) [GCC 6.3.0 20170118]
```

### CODE REVIEW

**Comment by gdelpierre** [comment]
cc @pilou- @Spredzy

File: lib/ansible/modules/messaging/kafka_topics.py [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+                        choices=['list', 'describe', 'alter', 'create', 'delete'],
+                        type='str'),
+        topic = dict(required=False, type='list'),
+        config = dict(required=False, type='dict'),
+        zookeeper = dict(required=True, type='list'),
+        replication_factor = dict(required=False, type='int'),
+        partitions = dict(required=False, type='int'),
+        kafka_path = dict(required=False, default='/opt/kafka',
+                                type='path'),
+        jaas_auth_file = dict(required=False, default='jaas.conf', type='str'),
```
Comment (mscherer) [inline lib/ansible/modules/messaging/kafka_topics.py:327]: Shouldn't it be 'path' here too ?

**Comment by jwoodrich** [comment]
@gdelpierre I was looking at implementing a similar module for Kafka topics and saw this pull request.  I think you've got a great idea, and thought I'd offer some feedback if you're open to it.  

I noticed that your modules aren't idempotent.  The topic and acl modules both have an action parameter, and the configs module uses add_configs and del_configs, which is also an action.  Not only would it be really useful if it was idempotent, the [Ansible documentation](https://docs.ansible.com/ansible/latest/playbooks_intro.html) indicates that modules should be idempotent.  Maybe instead of an action you could provide a state parameter to ensure that the topic, acl, or configuration is in one of two states: present or absent.  In this case, the module itself would determine if an add, update, or delete is needed, and make sure the entity actually is as described.  

Also, I tried running your module in a test environment that's not using jaas.  Do you think it might be too restrictive to require jaas for creating/modifying/deleting topics, given Kafka doesn't have this restirction?

**Comment by gdelpierre** [comment]
Separate onto 3 new PRs (one for each new modules, according to ansibot)

**Pull request closed without merging**